🎄🎅🏻 Santa Is Bringing KeyCloak to Town 🎄🎅🏻

jarmee opened the pull request:
- adds keycloak as service to docker-compose
- adds realm import to keycloak service
- updates documentation
- optimizes layout of layout component
- adds logout button to layout component
- adds [angular-oauth2-oidc](https://github.com/manfredsteyer/angular-oauth2-oidc) to client
  application
- adds environments to client application
- adds authconfig to environments
- formats files
- adds logic so that the content of the client app is only
 shown when the user is logged id
- adds cors config to api
- adds nest-keycloak-connect to api
- adds config for keycloak to api
- introduces apiUrl to environment of client
- loads stacks from api within stacks overview
- adds userName to layout component
- adds deployment job for keycloak docker container

Closed #55

jarmee-asol committed:
feat: adds authentication for client (#55)

- adds keycloak as service to docker-compose
- adds realm import to keycloak service
- updates documentation
- optimizes layout of layout component
- adds logout button to layout component
- adds [angular-oauth2-oidc](https://github.com/manfredsteyer/angular-oauth2-oidc) to client
  application
- adds environments to client application
- adds authconfig to environments
- formats files

jarmee-asol committed:
feat: adds authentication for client (#55)

- adds logic so that the content of the client app is only
 shown when the user is logged id
- adds cors config to api
- adds nest-keycloak-connect to api
- adds config for keycloak to api
- introduces apiUrl to environment of client
- loads stacks from api within stacks overview
- adds userName to layout component

jarmee-asol committed:
feat: adds deployment of keycloak (#55)

- adds deployment job for keycloak docker container

florianehmke commented on apps/api/src/app/app.module.ts:
```diff
@@ -11,6 +18,12 @@ import { StackTechnology } from './api/stacks/technologies';
+      authServerUrl: 'http://localhost:8081',
+      realm: 'my-stack',
+      clientId: 'api',
+      secret: 'Fhn3Me07pFQmT2y7pD401SXDE8qAObzf',
```
Warum eigentlich ein private client für das Backend? 
Ich hätte intuitiv jetzt einen public client gewählt und die API fungiert nur als resource server und validiert die Tokens.

Sonst würde ich es hier bevorzugen wenn das secret schon aus der env (über die ConfigService) kommt ;).

jarmee replied:
Die library `nest-keycloak-connect` unterstützt scheinbar nur `private` clients. Sollten wir dann noch im Nachgang ändern. Ich habe bereits ein Ticket angelegt, s. #57. Des Weiteren habe ich die gewünschten Änderungen vorgenommen.

jarmee-asol committed:
fix: adds missing export of client api (#55)

jarmee-asol committed:
fix: changes registering of KeycloakConnectModule to async (#55)

- changes KeycloakConnectModule to make use of registerAsync, so that needed
  configuration is loaded with nest.js config service.
- introduces four new environment variables to host the keycloak config (AUTH_SERVER_URL,
  AUTH_REALM, AUTH_CLIENT_ID, AUTH_SECRET)

jarmee-asol committed:
fix: activates silentRefresh of access token in client (#55)

jarmee-asol committed:
fix: removes unecessary char from authentication docs (#55)
